src/main/java/graphql/schema/DataFetchingEnvironment.java

```diff
@@ -44,11 +45,9 @@
      * Returns a context argument that is set up when the {@link graphql.GraphQL#execute(String, Object)} method
      * is invoked
      *
-     * @param <T> you decide what type it is
-     *
      * @return a context object
      */
-    <T> T getContext();
+    ExecutionContext getContext();
```

Comment:
> Convince me?
Specifically, I need access to the fragment definitions so I can inspect which fields will be resolved. Without the definitions I can only look at a FragmentSpread object, which doesn't tell me much.

For me, inspecting these queries is important because I have a schema that is built at runtime by inspecting my database ORM. Associations between my domain objects are lazily loaded, so a deep graph could cause many recursive queries to the db. If I can inspect the query then I can eagerly join these associations when the query is received and avoid N+1 problems.

> Needs to not break API
Right. I got ahead of myself. I think a way to fix it and not expose the `ExecutionContext` would be to pass `fragmentsByName` or a `List<SelectionSet>` that has no `FragmentSpread` objects as selections. The former is probably easiest.